lib/array_intersection.rb

```diff
@@ -1,6 +1,26 @@
+require "pry"
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Time complexity is O(n) (or, O(n+m)) we are looping through one array O(n), then searching worst case for a look up a hash table O(n)
+# Space complexity: O(n) with n elements in the hash table.
 def intersection(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return []
+  else
+    return array1.length >= array2.length ? sorting(array1, array2) : sorting(array2, array1)
+  end
+end
+
+def sorting(longer, shorter)
+  holder_hash = Hash.new
```

Comment:
`holder_hash` and `intersec` are extra datastructures that increase **space** complexity. Can you think of a better approach that doesn't use extra datastructures and has a **space** complexity of `O(1)`? (Hint: it may have _worse_ **time** complexity than your current one.)